### PR TITLE
Move repository create to RepositoryService

### PR DESCRIPTION
**Pull request by dmjb**
This completes the current refactoring of repository logic.

One small behaviour change is introduced: when the create logic is unable to insert the new repo into the DB, it will attempts to delete the webhook which it created. The previous logic would just leave the webhook in the customer's github repo.

# Summary

***Provide a brief overview of the changes and the issue being addressed. 
Explain the rationale and any background necessary for understanding the changes. 
List dependencies required by this change, if any.***

Fixes #(related issue)

## Change Type

***Mark the type of change your PR introduces:***

- [ ] Bug fix (resolves an issue without affecting existing features)
- [ ] Feature (adds new functionality without breaking changes)
- [ ] Breaking change (may impact existing functionalities or require documentation updates)
- [ ] Documentation (updates or additions to documentation)
- [x] Refactoring or test improvements (no bug fixes or new functionality)

# Testing

***Outline how the changes were tested, including steps to reproduce and any relevant configurations. 
Attach screenshots if helpful.***

# Review Checklist:

- [x] Reviewed my own code for quality and clarity.
- [x] Added comments to complex or tricky code sections.
- [ ] Updated any affected documentation.
- [x] Included tests that validate the fix or feature.
- [x] Checked that related changes are merged.

### CODE REVIEW

**Comment by dmjb** [comment]
Have tested these changes locally.

File: internal/repositories/github/service.go [hunk 1]
```diff
@@ -70,6 +92,63 @@ func NewRepositoryService(
...
+
+	// publish a reconciling event for the registered repositories
+	if err = r.pushReconcilerEvent(pbRepo, projectID); err != nil {
+		return nil, err
+	}
+
+	// Telemetry logging
+	logger.BusinessRecord(ctx).Provider = ghprovider.Github
+	logger.BusinessRecord(ctx).Project = projectID
+	logger.BusinessRecord(ctx).Repository = dbID
```
Comment (evankanderson) [inline internal/repositories/github/service.go:147]: Can we move these up earlier in the function to associate more errors with projects and repos?

File: internal/repositories/github/service.go [hunk 2]
```diff
@@ -134,3 +213,75 @@ func (r *repositoryService) deleteRepository(ctx context.Context, client ghclien
 
 	return nil
 }
+
+func (r *repositoryService) pushReconcilerEvent(pbRepo *pb.Repository, projectID uuid.UUID) error {
+	log.Printf("publishing register event for repository: %s/%s", pbRepo.Owner, pbRepo.Name)
```
Comment (evankanderson) [inline internal/repositories/github/service.go:218]: Again, `zerolog.Ctx(ctx)` will give us additional information, and we should prefer it in new code.

**Review by evankanderson — APPROVED** [review]
A couple minor points, but nothing blocking.

One other note: I'm slightly nervous about rewriting all the `handlers_repositories_tests.go` tests at the same time as refactoring the inner workings.  Showing that the existing tests pass (or what the behavior changes are) feels more robust in terms of ensuring that the behavior doesn't accidentally change due to the refactor.  But this is probably a good discussion for Discord, so I'll bring it there.

**Comment by dmjb** [comment]
@evankanderson Thanks for the feedback. I will incorporate your suggestions into a follow up PR.

**Pull request merged**